src/main/kotlin/org/rust/lang/core/types/RustEnumType.kt

```diff
@@ -4,12 +4,21 @@ import com.intellij.codeInsight.completion.CompletionUtil
 import org.rust.lang.core.psi.RustEnumItemElement
 import org.rust.lang.core.types.visitors.RustTypeVisitor
 
-class RustEnumType(enum: RustEnumItemElement) : RustStructOrEnumTypeBase() {
+class RustEnumType(
+    enum: RustEnumItemElement,
+    typeArguments: List<RustType> = emptyList()
+) : RustStructOrEnumTypeBase(typeArguments) {
```

Comment:
Hm, `StructType` is basically `SubstitutedStructType` with an empty substitution. Again, I don't see any concrete benefits from agregators/decorators, but I do see an increase in code size.